webapp/components/wpt-insights.js

```diff
@@ -19,66 +20,154 @@ class Insights extends ProductInfo(WPTFlags(PolymerElement)) {
-              [[item.query]]
-            </a>
-          </info-banner>
-          <p>
-            [[item.desc]]
-          </p>
-        </div>
-      </paper-card>
-    </template>
+
+    <wpt-anomalies></wpt-anomalies>
+    <wpt-flakes></wpt-flakes>
 `;
   }
 
   static get is() {
     return 'wpt-insights';
   }
+}
+window.customElements.define(Insights.is, Insights);
+
+const cardStyle = html`
+  <style>
+    paper-card {
+      display: block;
+      margin-top: 1em;
+      width: 100%;
+    }
+  </style>
+`;
 
-  get insights() {
-    const browsers = ['chrome', 'edge', 'firefox', 'safari'];
-    const anomalies = browsers.map(b => {
-      const others = browsers.filter(o => o !== b);
-      const othersPassing = others
-        .map(o => `(${o}:pass|${o}:ok)`)
-        .join(' ');
-      const query = `!${b}:pass !${b}:ok ${othersPassing}`;
-      const url = new URL('/results/', window.location);
-      url.searchParams.set('q', query);
-      return {
-        name: `${this.displayName(b)}-only failures`,
-        url,
-        query,
-        desc: `Tests that are failing in ${this.displayName(b)}, but passing in the other browsers (${others.map(this.displayName).join(', ')})`,
-      };
-    });
-    const flakes = browsers.map(b => {
-      const query = `(${b}:pass|${b}:ok) (${b}:timeout|${b}:error|${b}:fail)`;
-      const url = new URL('/results/', window.location);
-      url.searchParams.set('q', query);
-      url.searchParams.set('product', b);
-      url.searchParams.set('max-count', 10);
-      url.searchParams.set('labels', 'master,experimental');
-      return {
-        name: `Flakes in the last 10 ${this.displayName(b)} runs`,
-        url,
-        query,
-        desc: `Tests that have both passing and non-passing results in the last 10 ${this.displayName(b)} runs`,
-      };
-    });
-    return anomalies.concat(flakes);
+class Flakes extends ProductInfo(PolymerElement) {
+  static get is() {
+    return 'wpt-flakes';
+  }
+
+  static get template() {
+    return html`
+    ${cardStyle}
+    <paper-card>
+      <div class="card-content">
+        <h3>Flakes</h3>
+        <browser-picker browser="{{browser}}"></browser-picker>
+        <info-banner>
+          <a class="query" href="[[url]]">[[query]]</a>
+        </info-banner>
+        <p>
+          Tests that have both passing and non-passing results in the last 10 [[browserDisplayName]] runs
+        </p>
+      </div>
+    </paper-card>
+`;
+  }
+
+  static get properties() {
+    return {
+      browser: String,
+      browserDisplayName: {
+        type: String,
+        computed: 'displayName(browser)',
+      },
+      query: {
+        type: String,
+        computed: 'computeQuery(browser)',
+      },
+      url: {
+        type: URL,
+        computed: 'computeURL(browser, query)',
+      }
+    };
+  }
+
+  computeQuery(browser) {
+    return `(${browser}:pass|${browser}:ok) ${browser}:!pass ${browser}:!ok`;
```

Comment:
That means "there exists a `browser` run that is either `PASS` or `OK`, and there exists a browser that is not `PASS`, and there exists a browser that is not `OK`". I don't think that's what you want.

Reply:
Unfortunately, because we don't have careful expansion of some kind of `passing` token that means "`PASS` or `OK`", I think you need `(${browser:pass|${browser}:ok) (${browser}:[each non-pass-or-ok value]|[...])`

Reply:
Even with the parens? I'd expect that the parens equate it to "there exists a `browser` run that is not pass and is not ok". 
Each atom at the root should be a separate exists, no?

Reply:
Will revert this and discuss the meaning of parens in another thread.